lib/pushmi_pullyu/aip.rb

```diff
@@ -0,0 +1,40 @@
+require 'pushmi_pullyu'
+require 'bagit'
+require 'archive/tar/minitar'
+
+module PushmiPullyu::AIP
+  # Exceptions
+  class BagInvalid < StandardError; end
+  class NoContentFilename < StandardError; end
+  class FedoraFetchError < StandardError; end
+  class SolrFetchError < StandardError; end
+
+  def self.create(noid, should_skip_download: false, should_clean_work_directories: true)
```

Comment:
I'm being really paranoid, but can we add a bit of validation here to verify that `noid` isn't empty? The queue should never yield an empty string, but if it did, the logic of `aip_directory` will lead to the wrong directory being `rm -rf`'d. This extends to a few other places in the codebase where an empty string and a badly chosen or missing `work_dir` could lead to accidentally deleting the wrong directory